iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -4,1310 +4,1043 @@
+            Logging.Info(this, $"Received twin patch response for request id {requestId} with status {patchTwinResponse.Status}.");
+            if (patchTwinResponse.Status != 204)
             {
-                // Wait to grab the semaphore, and then unsubscribe from C2D messages and set _isDeviceReceiveMessageCallbackSet  to false.
-                // Once _isDeviceReceiveMessageCallbackSet  is set to false, all received C2D messages can be returned via the polling ReceiveMessageAsync() call.
-                await _deviceReceiveMessageSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
-
-                try
-                {
-                    // The TransportState is transitioned to Receiving only if the device is subscribed to _deviceboundMessageFilter.
-                    // Only if the subscription has been previously set, we will send the unsubscribe packet.
-                    if (State == TransportState.Receiving
-                        && TryStateTransition(TransportState.Receiving, TransportState.Open))
-                    {
-                        // Update the TransportState from Receiving to Open.
-                        await _channel.WriteAsync(new UnsubscribePacket(0, _deviceboundMessageFilter)).ConfigureAwait(true);
-                    }
-                    _isDeviceReceiveMessageCallbackSet = false;
-                }
-                finally
-                {
-                    _deviceReceiveMessageSemaphore.Release();
-                }
+                //TODO tim
+                throw new IotHubClientException("Failed to send twin patch");
             }
-            finally
-            {
-                if (Logging.IsEnabled)
-                    Logging.Exit(this, cancellationToken, nameof(DisableReceiveMessageAsync));
-            }
-        }
-
-        public override async Task EnableMethodsAsync(CancellationToken cancellationToken)
-        {
-            cancellationToken.ThrowIfCancellationRequested();
-            EnsureValidState();
 
-            await _channel.WriteAsync(new SubscribePacket(0, new SubscriptionRequest(MethodPostTopicFilter, _qosReceivePacketFromService))).ConfigureAwait(true);
+            //TODO new twin version should be returned here, but API surface doesn't currently allow it
+            //return patchTwinResponse.Version;
         }
 
-        public override async Task DisableMethodsAsync(CancellationToken cancellationToken)
+        protected override void Dispose(bool disposing)
         {
-            cancellationToken.ThrowIfCancellationRequested();
-            EnsureValidState();
-
-            await _channel.WriteAsync(new UnsubscribePacket(0, MethodPostTopicFilter)).ConfigureAwait(true);
+            base.Dispose(disposing);
+            _mqttClient?.Dispose();
+
+            // TODO notify the user for these that they failed? Clear these when closing instead?
```

Comment:
Log them out, maybe.

Reply:
Got rid of these collections, so this is now moot